src/compas_timber/fabrication/btlx.py

```diff
@@ -51,113 +43,166 @@ class BTLx(object):
             ),
         ]
     )
+    FILE_HISTORY_ATTRIBUTES = OrderedDict(
+        [
+            ("CompanyName", "Gramazio Kohler Research"),
+            ("ProgramName", "COMPAS_Timber"),
+            ("ProgramVersion", "Compas: {}".format(compas.__version__)),
+            ("ComputerName", "{}".format(os.getenv("computername"))),
+            ("UserName", "{}".format(os.getenv("USERNAME"))),
+            ("FileName", ""),
+            ("Date", "{}".format(date.today())),
+            ("Time", "{}".format(datetime.now().strftime("%H:%M:%S"))),
```

Comment:
since this is a class attribute (which I don't have a problem with) these two should be formatted at write time.
otherwise several consequent write operations will all have the same timestamp.
you can basically leave the placeholder strings `"{}"` there and call `.format()` on them later